src/renderer/components/recording-page/recording-visualization.tsx

```diff
@@ -2,134 +2,162 @@ import log from 'electron-log';
 import React, { FC, useEffect, useRef } from 'react';
 import Col from 'react-bootstrap/Col';
 import Row from 'react-bootstrap/Row';
-import useInterval from 'react-use/lib/useInterval';
 import WaveSurfer from 'wavesurfer.js';
+import MicrophonePlugin from 'wavesurfer.js/dist/plugin/wavesurfer.microphone';
 import SpectrogramPlugin from 'wavesurfer.js/dist/plugin/wavesurfer.spectrogram';
 
 import { noOp } from '../../../common/env-and-consts';
-import { checkFileExistence, join, readFile } from '../../utils';
+import { acquireAudioInputStream, checkFileExistence, join, readFile } from '../../utils';
+import { FileMonitor } from '../../utils/file-monitor';
+import { useAudioInputOutputDevices } from '../settings-page/hooks';
 
 import { State } from './types';
 
 interface RecordingVisualizationProps {
   prevState?: State;
   state: State;
   filePath?: string;
+  basePath: string;
```

Comment:
There are many fields in this interface. Add TSDoc to these fields.

Reply:
Instead of using `@property`, declare the documentations above the properties themselves, e.g.:
```
/**
 * File path of the current project, used for the {@link FileMonitor}.
 */
filePath?: string;

/**
 * Name of the current recording item, used to limit the {@link FileMonitor} events
 */
basePath: string;
```